### DISCUSSION TITLE
Lesson 7: Make - make: forge: Permission denied

### DISCUSSION THREAD

**Discussion by tristanpeter** [Cyfrin/foundry-full-course-cu#936] [General]
Hi All,

I have searched through the discussions, the issues, Googled this and ChatGPT'd it, but so far I have not found an understandable explanation as to why I am getting the below error when I try to run "make build" in my WSL terminal. 
```
$ make build
forge build
make: forge: Permission denied
make: *** [Makefile:3: build] Error 127
```

Here are my Makefile contents:

```
-include .env

build:; forge build
a:; git --version
format:; forge fmt

```

The command "make a" returns the git version correctly, but "make build" and "make format" both return this error:

```

make: forge: Permission denied
make: *** [Makefile:5: format] Error 127

```

Any ideas? 

Thanks all!
Tristan

**Comment by GenevaRoc** [comment]
Can I see your code?

**Reply by tristanpeter** [reply]
@GenevaRoc any code specifically, or all of it? Thanks for your help!

**Reply by tristanpeter** [reply]
...from Lesson 7 I mean :)

**Comment by tristanpeter** [comment]
Hey All,

I figured the out the problem. If anyone has this issue, it happened because I was running the make commands in my WSL terminal, but I only had Foundry/Forge installed in my bash terminal, so the make commands were not recognised. I installed Foundry on wsl and all's good again(for now at least). This is probably a basic mistake to have made on my part, but I must admit, I find these different terminals, and how they interact with each other a bit confusing. You live you learn! :)
Go raibh maith agat
Tristan
